src/handleAction.js

```diff
@@ -7,19 +7,21 @@ export default function handleAction(type, reducers, defaultState) {
     ? type.toString()
     : type;
 
+  const [nextReducer, throwReducer] = isFunction(reducers)
+    ? [reducers, reducers]
+    : [reducers.next, reducers.throw];
+
+  if (nextReducer === undefined || throwReducer === undefined) {
+    throw new TypeError(
+      'reducers argument must be either a function or an object of shape {next, throw}'
+    );
+  }
+
   return (state = defaultState, action) => {
     // If action type does not match, return previous state
     if (action.type !== typeValue) return state;
 
-    const handlerKey = action.error === true ? 'throw' : 'next';
-
-    // If function is passed instead of map, use as reducer
-    if (isFunction(reducers)) {
-      reducers.next = reducers.throw = reducers;
```

Comment:
Yeah. That was the original problem I saw that compelled me to make this fix– mutating/assigning to the input parameter seemed like a VERY bad idea.